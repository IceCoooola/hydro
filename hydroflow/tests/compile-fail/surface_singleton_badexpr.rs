/// Correct reference but using it wrong (bad expression).
fn main() {
    let mut df = hydroflow::hydroflow_syntax! {
        my_ref = source_iter(15_u32..=25) -> fold(|| 0, |a, b| *a = std::cmp::max(*a, b));
        source_iter(10_u32..=30)
            -> persist()
            -> filter(|value| value <= #my_ref)
            -> null();

    };
    df.run_available();
}
