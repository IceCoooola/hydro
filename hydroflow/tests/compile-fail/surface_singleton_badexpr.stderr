error[E0308]: mismatched types
 --> tests/compile-fail/surface_singleton_badexpr.rs:7:40
  |
7 |             -> filter(|value| value <= #my_ref)
  |                                        ^^^^^^^ expected `&u32`, found integer
  |
help: consider removing deref here
  |
7 |             -> filter(|value| value <= #my_ref)
  |
